python/ray/monitor.py

```diff
@@ -49,7 +49,11 @@ def parse_resource_demands(resource_load_by_shape):
                 backlog_queue = infeasible_bundles
             else:
                 backlog_queue = waiting_bundles
-            for _ in range(resource_demand_pb.backlog_size):
+            for _ in range(
+                    # Limit the backlog queue size to reduce the overhead of
+                    # the resource demand scheduler.
+                    min(ray_constants.MAX_BACKLOG_SIZE,
+                        resource_demand_pb.backlog_size)):
```

Comment:
Can we do `MAX_BACKLOG_SIZE - len(backlog_queue)` to bound the entire vector rather than the just the portion from the backlog?

Reply:
Can we actually not? Since it is reduced to 1k, it should handle up to say 20 different types. I am worried this might fill the demand with the first type (say just CPUs) and sometimes it is better to allow a mix of CPUS and GPUS.

Reply:
do you mean `MAX_BACKLOG_SIZE - len(waiting_bundles + infeasible_bundles)`?